message/validation/alan/signed_ssv_message.go

```diff
@@ -0,0 +1,173 @@
+package msgvalidation
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	spectypes "github.com/bloxapp/ssv-spec/alan/types"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"golang.org/x/exp/slices"
+
+	"github.com/bloxapp/ssv/network/commons"
+	ssvmessage "github.com/bloxapp/ssv/protocol/v2/message"
+)
+
+func (mv *messageValidator) decodeSignedSSVMessage(pMsg *pubsub.Message) (*spectypes.SignedSSVMessage, error) {
+	signedSSVMessage := &spectypes.SignedSSVMessage{}
+	if err := signedSSVMessage.Decode(pMsg.GetData()); err != nil {
+		e := ErrMalformedPubSubMessage
+		e.innerErr = err
+		return nil, e
+	}
+
+	return signedSSVMessage, nil
+}
+
+func (mv *messageValidator) validateSignedSSVMessage(signedSSVMessage *spectypes.SignedSSVMessage) error {
+	if signedSSVMessage == nil {
+		return ErrEmptyPubSubMessage
+	}
+
+	signers := signedSSVMessage.GetOperatorIDs()
+
+	if len(signers) == 0 {
+		return ErrNoSigners
+	}
```

Comment:
I've also added the check that it's not bigger than the committee size, as we discussed. Is it added here somewhere else?

Reply:
https://github.com/ssvlabs/ssv/pull/1374#discussion_r1580584425